core/client/verify.go

```diff
@@ -101,46 +98,36 @@ func (v *Verifier) VerifyGetEntryResponse(ctx context.Context, domainID, appID,
 	Vlog.Printf("✓ VRF verified.")
 
 	leafProof := in.GetLeafProof()
-	if leafProof == nil {
+	if leafProof.GetLeaf() == nil {
 		return ErrNilProof
 	}
+	leafProof.Leaf.Index = index[:]
 
-	leaf := leafProof.GetLeaf().GetLeafValue()
-	proof := leafProof.GetInclusion()
-	expectedRoot := in.GetSmr().GetRootHash()
-	mapID := in.GetSmr().GetMapId()
-	if err := merkle.VerifyMapInclusionProof(mapID, index[:], leaf, expectedRoot, proof, v.hasher); err != nil {
+	if err := v.mapVerifier.VerifyMapLeafInclusion(in.GetSmr(), leafProof); err != nil {
```

Comment:
Go is pretty opinionated about capitalisation of initialisms - `GetSmr` should be `GetSMR`.

Reply:
`GetSmr` is unfortunately generated directly by protoc from "smr" in the proto file.